bootstrap/secret/secure.go

```diff
@@ -177,6 +178,9 @@ func (p *SecureProvider) StoreSecret(path string, secrets map[string]string) err
 		return err
 	}
 
+	// Execute Callbacks on registered secret paths.
+	p.SecretsUpdatedWithPath(path)
+
```

Comment:
See my earlier point about not exposing this via the public interface and supplying the new secret value since we have it anyway.

Reply:
Per other comments this sadly needs to be public. I.e. this interface wasn't originally design to be used directly by the custom App and Device services. We'll fix it in 3.0.